Added support for running migrations before rehydrating state

adam-golab opened the pull request:
Discussion about the desired feature can be found under the #4 issue.

TODO:
[X] add documentation to the readme

Commit:
feat: added support for running migrations before rehydrating state

ThibautMarechal commented on src/reduxPhoenix.js:
```diff
@@ -47,17 +84,24 @@ export default function persistStore(store, {
     }
     const persistedValue = deserialize(persistedJson);
-    const { persistedState, saveDate } = persistedValue || {};
+    const { persistedState, saveDate, migrations: appliedMigrations } = persistedValue || {};
```
You do not set "migrations" when saving into the storage

adam-golab replied:
It is saved. Please take a look on line 137

ThibautMarechal replied:
It was and old comment not submitted directly :)

adam-golab replied:
yeah, I forgot about it yesterday ;)

Commit:
fix: added missing tests for saving applied migrations

Commit:
refactor: use functional programing paradigm

ThibautMarechal reviewed (changes requested):
First of all thanks ! It worked well for my use case.

Maybe it we should add a validation for the migrations objects.

The code may throw in the migrations (the classic Cannot read property of '...' ) 
Maybe a try / catch around the migrations can catch the error ?

Commit:
docs: added documentation for migrations feature

Commit:
fix: added null checks

adam-golab commented:
@ThibautMarechal good idea. I added some null checks a moment ago. I don't think that wrapping it in a `try/catch` block is necessary here. 

But I added a few checks that only migrations with the `name` and `up/down` method will be applied

Commit:
fix: build

Commit:
1.3.0

adam-golab commented:
Published as a version: 1.3.0
